src/auth/feedback-secret.guard.ts

```diff
@@ -0,0 +1,24 @@
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  UnauthorizedException,
+} from '@nestjs/common';
+import { Env } from 'src/env';
+
+@Injectable()
+export class FeedbackSecretGuard implements CanActivate {
+  constructor() {}
+
+  canActivate(context: ExecutionContext): boolean {
+    const request = context.switchToHttp().getRequest();
+    const secret = Env.SECRET.FEEDBACK;
+    const requestSecret = request.headers['x-feedback-secret'];
+
+    if (requestSecret && requestSecret === secret) {
+      return true;
+    } else {
+      throw new UnauthorizedException('Invalid feedback secret');
+    }
```

Comment:
PrimathonAICodeReviewer

It's recommended to log a warning or error message instead of throwing an exception directly. This provides better visibility into unauthorized attempts.

```diff
-       throw new UnauthorizedException('Invalid feedback secret');
+       console.warn('Invalid feedback secret provided');
```

### Review Feedback

The code changes look good overall, with minor suggestions for improvement as highlighted above.


<!-- This is an auto-generated comment by OSS PrimathonAICodeReviewer -->